cleaner/custom_pre/classes/clean.php

```diff
@@ -0,0 +1,45 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * @package    cleaner_custom_pre
+ * @copyright  2019 Catalyst IT
+ * @author     Srdjan Janković <srdjan@catalyst.net.nz>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace cleaner_custom_pre;
```

Comment:
I'd rename the plugin so the intent is more obvious. 'Custom task' doesn't tell you much, but 'Custom SQL' tells you what it is for. I would not mention 'pre' at all, almost all tasks are pre wash so it's not adding much value, unless you added a second cleaner which was post wash.

This has been on the roadmap for a while, so in your commit can you please reference ```#51```

https://github.com/catalyst/moodle-local_datacleaner/issues/51

Reply:
I'd just call this 'sql', ie cleaner/sql/classes and 'cleaner_sql'

Reply:
Done I believe. Named `custom_sql_{pre|post}`. Can drop custom_ if you wish. pre and post are required, for both phases.